app/services/submission_service.rb

```diff
@@ -87,6 +89,10 @@ def update_submission(
           submission.assign_attributes(
             reject_non_target_supply_artist(submission.artist_id)
           )
+          if submission.submitted? &&
+               submission.user.save_submission_to_my_collection?
```

Comment:
**suggestion:** We should check that the submission was made by an authenticated user before creating an artwork in My Collection. For anonymous users, the artwork will be created once they create an Artsy account.

Reply:
if the user is not authorized, then `submission.user&.save_submission_to_my_collection?`  return nil

Reply:
Got it. Thanks for adding the safe navigation operator (`&.`) 🙌

Reply:
That's true for our current configuration. However, one day we will remove the feature flag and all submissions will be allowed to save submissions to My Collection. When that happens, we will need to remember that submissions from anonymous users should not be added to My Collection, so I think it's better if we prepare for that now.

Reply:
Hmm, as far as I remember, if this is an anonymous submission, then it does not have a user, that is, in any case, submission.user == nil ==> submission.user&.save_submission_to_my_collection? == nil

Reply:
That is all correct, and will remain true until we launch the My Collection integration.

Once we launch the My Collection integration, we will remove the `swa_my_collection` feature flag as well as the `save_submission_to_my_collection?` method because we won't need them anymore.

When that happens, we will need to add an additional check for anonymous users. What I'm proposing is that we add the check for anonymous users today, so that we don't forget to add that when we remove the feature flag in the future.